ceph/ceph_admin/__init__.py

```diff
@@ -215,6 +215,18 @@ def setup_upstream_repository(self, repo_url=None):
                 check_ec=False,
             )
 
+            # public repo: needed to compensate for dependencies required during
+            # installation of ceph-common and other pkg RPMs
+            public_repo_url = (
+                f"https://dl.fedoraproject.org/pub/epel/"
+                f"{node.distro_info['VERSION_ID'][0]}/Everything/x86_64/"
+            )
+            node.exec_command(
+                sudo=True,
+                cmd=f"yum-config-manager --add-repo {public_repo_url}",
+                check_ec=False,
+            )
+
```

Comment:
can we create a separate commit for this change?

IMHO, we should have a 
- a commit modifying fetch_artifact method
- a commit for installing `ceph-common` packages.

Reply:
hi @psathyan I believe the ceph-common packages to be installed on all nodes as we are not fully utilising the sos-report log collection was informed and was ACK'ed by CI team. Do you need us to add that support?

Reply:
There are two sets of changes in a single commit. The ask here is to have them in separate commits for better change log.

Reply:
@psathyan is it okay to include both the commits with this PR? Or would you like me to raise a separate PR for the client package installation change?

Reply:
Same PR should be fine.

Reply:
@psathyan I have added two separate commits